querytest/compare.go

```diff
@@ -0,0 +1,222 @@
+package querytest
+
+import (
+	"errors"
+	"fmt"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/influxdata/flux"
+	"github.com/influxdata/flux/execute"
+	"github.com/influxdata/flux/values"
+)
+
+// FloatTolerance represents a sufficient distance that all
+// float values must be within in order to be considered equal.
+const FloatTolerance float64 = 1e-50
+
+// FloatULP is the maximum number of floats that are allowed to lie
+// between two floating point values and still be considered equal.
+const FloatULP uint = 2
+
+type tables struct {
+	name string
+	data flux.ResultIterator
+}
+
+func CreateTables(name string, data flux.ResultIterator) *tables {
+	return &tables{
+		name: name,
+		data: data,
+	}
+}
+
+type results struct {
+	equal bool
+	err   error
+	fmt   map[string]*execute.Formatter
+}
+
+func (r *results) Equal() bool {
+	return r.equal
+}
+func (r *results) Err() error {
+	return r.err
+}
+func (r *results) Fmt() map[string]*execute.Formatter {
+	return r.fmt
+}
+
+func (t *tables) EqualWithTolerance(s *tables, tol float64, ulp uint) *results {
```

Comment:
I gave `tables` a name param which the caller provides, so that the caller doesn't have to remember if "want" should be on the left or right. See https://github.com/influxdata/platform/pull/1450/files#diff-afa3e17b057923f1e8fcb59b758370ddR273 for usage. Perhaps I should explain that here?

Reply:
Ah, okay.  That makes sense.

It would be good to add comments to all the exported items here, especially since this code could also be used by contributors as they add new functions to Flux.

Reply:
Agreed, I'll get get these comments up with the rest of the changes.